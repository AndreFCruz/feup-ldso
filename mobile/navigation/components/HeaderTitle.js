import React from 'react';
import { Image, StyleSheet } from 'react-native';
import PropTypes from 'prop-types';

class HeaderTitle extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            darkLogo: require('../../assets/images/impactup.png'),
            whiteLogo: require('../../assets/images/impactup-white.png')
        };
    }

    render() {
        const logo = this.props.invertedMode ? this.state.whiteLogo : this.state.darkLogo;

        return (
            <Image
                source={logo}
                style={styles.image}
            />
        );
    }
}

HeaderTitle.propTypes = {
    invertedMode: PropTypes.bool
};

const styles = StyleSheet.create({
    image: {
        width: 120,
        height: 30
    },
});

export default HeaderTitle;
