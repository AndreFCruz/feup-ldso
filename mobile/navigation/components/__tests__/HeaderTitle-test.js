import 'react-native';
import React from 'react';
import HeaderTitle from '../HeaderTitle';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

it('renders logo', () => {
    const wrapper = shallow(<HeaderTitle invertedMode={false} />);

    expect(wrapper).toMatchSnapshot();
});

