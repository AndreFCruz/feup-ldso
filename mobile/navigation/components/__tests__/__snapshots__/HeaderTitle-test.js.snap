// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders logo 1`] = `
<Image
  source={1}
  style={
    Object {
      "height": 45,
      "width": 45,
    }
  }
/>
`;
